pkg/api/validate/vm.go

```diff
@@ -13,7 +13,23 @@ import (
 // To add new instance types, needs Project Managment's involment and instructions are below.,
 // https://github.com/Azure/ARO-RP/blob/master/docs/adding-new-instance-types.md
 
-var SupportedMasterVmSizes = map[api.VMSize]api.VMSizeStruct{
+const VMRoleMaster string = "master"
+const VMRoleWorker string = "worker"
+
+var supportedVMSizesByRoleMap = map[string]map[api.VMSize]api.VMSizeStruct{
+	VMRoleMaster: supportedMasterVmSizes,
```

Comment:
Hey Srini. How are we validating that these sizes are *actually* supported for the master? I see we have an explicit list of them in vm.go, but whether or not masters can actually support this size is based on the subscription level allotment, regional availability (which may be dynamic) etc. Are we doing any specific checking inside these parameters?

Reply:
There's the sku validator that does this kind of this iirc. It's in the frontend